lib/rester.rb

```diff
@@ -19,6 +20,11 @@ def load_tasks
     def connect(*args)
       if (service = args.first).is_a?(Class) && service < Service
         Client.new(Client::Adapters::LocalAdapter.new(*args))
+      elsif args.first.is_a?(String) && Pathname(args.first).file?
+        unless (file_ext = Pathname(args.first).extname) == '.yml'
+          raise Errors::InvalidStubFileError, "Expected .yml, got #{file_ext}"
+        end
+        Client.new(Client::Adapters::StubAdapter.new(*args))
```

Comment:
Even better?

```ruby
def connect(service, params={})
  adapter_klass = Client::Adapters.list.find { |k| k.can_connect_to?(service) }
  fail "unable to connect to: #{service.inspect}" unless adapter_klass
  adapter = adapter_klass.new(service)
  Client.new(adapter, params)
end
```

Then in each adapter class, define a `can_connect_to?` method that will do whatever checks are necessary.

Also, define an `Client::Adapters.list` method that provides a list of all the adapters (this could be dynamic based on what's in the `adapters/` directory.